### PR TITLE
Autocomplete

### PR DESCRIPTION
**Pull request by benma**
Closes https://github.com/digitalbitbox/bitbox02-firmware/issues/578

### CODE REVIEW

File: src/ui/components/trinary_input_string.c [hunk 1]
```diff
@@ -235,6 +235,31 @@ static void _render(component_t* component)
...
+        bool is_prefix = strncmp(data->string, word, data->string_index) == 0;
+        if (is_prefix) {
+            if (found_word_idx != data->wordlist_size) {
+                // Not unique.
+                return;
+            }
+            found_word_idx = word_idx;
+        }
+    }
+    data->string_index =
```
Comment (benma) [inline src/ui/components/trinary_input_string.c:259]: could think about getting rid of string_index completely, and use `strlen(data->string)` in its place everywhere.

File: src/ui/components/trinary_input_string.c [hunk 2]
```diff
@@ -235,6 +235,31 @@ static void _render(component_t* component)
...
+// word.
+static void _maybe_autocomplete(component_t* trinary_input_string)
+{
+    data_t* data = (data_t*)trinary_input_string->data;
+    if (data->wordlist == NULL) {
+        return;
+    }
+    // initial value means no word was found yet.
+    size_t found_word_idx = data->wordlist_size;
+    for (size_t word_idx = 0; word_idx < data->wordlist_size; word_idx++) {
```
Comment (x1ddos) [inline src/ui/components/trinary_input_string.c:248]: I think there are ways to optimize this but probably not worth it today.

**Review by x1ddos — APPROVED** [review]
Cool. LGTM.

File: src/ui/components/trinary_input_string.c (hunk 2 above)
Comment (benma) [reply src/ui/components/trinary_input_string.c:248]: today or maybe ever? it's a small word list, even the whole prefix search to create the keyboard after each tap is not noticeable. or did you mean something else?

Comment (x1ddos) [reply src/ui/components/trinary_input_string.c:248]: yeah I actually meant "maybe ever" :)
didn't notice any issues visually.

**Pull request merged**